src/core/Children.h

```diff
@@ -37,7 +37,7 @@ class Children
   [[nodiscard]] const std::shared_ptr<const Context>& getContext() const { return context; }
 
 private:
-  const LocalScope *children_scope;
+  const std::shared_ptr<LocalScope> children_scope;
```

Comment:
Re-reading this PR: The LocalScope is meant to be const here, no its pointer, so I believe that here (and other places) you'd want to use `shared_ptr<const ...>` rather than `const shared_ptr<...`

Reply:
Aha! Though apparently it can also be a const shared_ptr in most places FWIW. Done.

Reply:
Yeah, the main thing to think about is that the object itself should not be changeable in most places. ..and in some places, the pointer itself can be constant too.